react/training/src/redux/Preferences/action.js

```diff
@@ -0,0 +1,49 @@
+import UserService from '~services/UserService';
+
+const privateActions = {
+  assignUserData: data => ({
+    type: 'USER_DATA_SUCCESS',
+    target: 'userData',
+    payload: data
+  }),
+  requestHasError: isError => ({
+    type: 'USER_DATA_FAILURE',
+    target: 'userData',
+    payload: isError
+  }),
+  requestUpdate: data => ({
+    type: 'USER_DATA_UPDATE',
+    target: 'userDataUpdated',
+    payload: data
+  })
+};
+
+const actionCreators = {
+  getUser: () => async dispatch => {
```

Comment:
Let's use fetch middleware here. later I explain to you in person because there is no documentation of this